### PR TITLE
ch4/ofi: remove setting size in dynamic window

### LINKED ISSUE #5148: MPI_WIN_SIZE attribute for dynamic windows in ch4:ofi
Consider the following trivial code

```python
from mpi4py import MPI
win = MPI.Win.Create_dynamic(MPI.INFO_NULL, MPI.COMM_SELF)
base = win.Get_attr(MPI.WIN_BASE)
size = win.Get_attr(MPI.WIN_SIZE)
win.Free()
print(base, size)
```

Depending on the device, I get the following output printed:

| Device | Output |
| --- | --- |
| ch3 | `0 0` |
| ch4:ucx | `0 0` |
| ch4:ofi | `0 -1` |

May I request to fix `ch4:ofi` to also return size := 0 ?

### PR DESCRIPTION
**Pull request by hzhou**
## Pull Request Description
Rather than modifying win->size for the purpose of fi_mr_reg, simply
special treat the case of dynamic window there.

The extra comment is fi_mr_reg is removed, as they are merely repeating
the documentation and creates clutter for reading code.

Fixes #5148 
[skip warnings]
## Author Checklist
* [x] **Provide Description** 
      Particularly focus on _why_, not _what_. Reference background, issues, test failures, xfail entries, etc.
* [x] **Commits Follow Good Practice**
      Commits are self-contained and do not do two things at once. 
      Commit message is of the form: `module: short description` 
      Commit message explains what's in the commit.
* [x] **Passes All Tests**
      Whitespace checker. Warnings test. Additional tests via comments.
* [x] **Contribution Agreement**
      For non-Argonne authors, check [contribution agreement](http://www.mpich.org/documentation/contributor-docs/). 
      If necessary, request an explicit comment from your companies PR approval manager.

### CODE REVIEW

**Comment by hzhou** [comment]
test:mpich/ch4/ofi

**Comment by hzhou** [comment]
test:mpich/ch4/ofi

**Comment by hzhou** [comment]
test:mpich/ch4/ofi

File: src/mpid/ch4/netmod/ofi/ofi_win.c [hunk 1]
```diff
@@ -148,18 +148,21 @@ static int win_allgather(MPIR_Win * win, void *base, int disp_unit)
...
+                len = UINTPTR_MAX - (uintptr_t) base;
+            } else {
+                len = (size_t) win->size;
+            }
+
+            MPIDI_OFI_CALL_RETURN(fi_mr_reg(MPIDI_OFI_global.ctx[vni].domain, base, len,
+                                            FI_REMOTE_READ | FI_REMOTE_WRITE, 0ULL,
+                                            MPIDI_OFI_WIN(win).mr_key, 0ULL,
+                                            &MPIDI_OFI_WIN(win).mr, NULL), rc);
+        }
```
Comment (minsii) [inline src/mpid/ch4/netmod/ofi/ofi_win.c:165]: Can we keep the original comment for each parameter of `fi_mr_reg`? They are useful. 

Besides, it will be better if we can avoid unnecessary changes (e.g., trying to change only the line of `win->size,  /* In:  Length              */` in the `fi_mr_reg` block. This is a very "sensitive" function. We often use `git blame` to check the reason of updates. So, let's avoid unnecessary change

Comment (hzhou) [reply src/mpid/ch4/netmod/ofi/ofi_win.c:155]: Alright. It is unfortunate that we have very different views on these practices.

**Comment by hzhou** [comment]
test:mpich/ch4/ofi

**Review by minsii — APPROVED** [review]
